src/talk/renderer/Viewer/ViewerApp.vue

```diff
@@ -0,0 +1,103 @@
+<!--
+  - @copyright Copyright (c) 2023 Grigorii Shartsev <me@shgk.me>
+  -
+  - @author Grigorii Shartsev <me@shgk.me>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<NcModal v-if="file"
+		:show.sync="isOpen"
+		class="viewer-modal"
+		:title="file.basename"
+		size="full"
+		:close-button-contained="false"
+		dark
+		@close="close">
+		<component :is="viewComponent"
+			v-if="viewComponent"
+			:file="file" />
+
+		<template #actions>
+			<NcActionLink :href="link">
+				<template #icon>
+					<MdiOpenInNew />
+				</template>
+				{{ t('talk_desktop', 'Open in a Web-Browser') }}
+			</NcActionLink>
+		</template>
+	</NcModal>
+</template>
+
+<script>
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcActionLink from '@nextcloud/vue/dist/Components/NcActionLink.js'
+import MdiOpenInNew from 'vue-material-design-icons/OpenInNew.vue'
+
+import { generateUrl } from '@nextcloud/router'
+
+const noop = () => {}
+
+export default {
+	name: 'ViewerApp',
+
+	components: {
+		MdiOpenInNew,
+		NcModal,
+		NcActionLink,
+	},
+
+	data() {
+		return {
+			isOpen: false,
+			onClose: noop,
+			file: null,
+		}
+	},
+
+	computed: {
+		viewComponent() {
+			return this.file && OCA.Viewer.availableHandlers.find((handler) => handler.mimes.includes(this.file.mime))?.component
+		},
+
+		link() {
+			return this.file && generateUrl(`/f/${this.file.fileid}`)
+		},
+	},
+
+	methods: {
+		open({ fileInfo, onClose = noop } = {}) {
+			this.onClose = onClose
+			this.file = fileInfo
+			this.isOpen = true
+		},
+
+		close() {
+			this.file = null
+
+			this.onClose()
+			this.onClose = noop
+		},
+	},
+}
+</script>
+
+<style scoped>
+.viewer-modal :deep(.modal-container) {
```

Comment:
To show RightSidebar (need to be applied to the div with class `modal-mask`):
`width: calc(100vw - clamp(300px, 27vw, 500px));`

Reply:
Maybe we don't need sidebar here...

Reply:
Then it would block a chat during the call

Reply:
But you cannot open an editable document anyway

Reply:
Let's keep it as-is then and see, if someone will complain later